src/foam/nanos/auth/InvoiceHistoryAuthorizer.js

```diff
@@ -0,0 +1,91 @@
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'InvoiceHistoryAuthorizer',
+  implements: ['foam.nanos.auth.Authorizer'],
+
+  documentation: `AuthorizableAuthorizer which checks if user has access to the invoice history record`,
+
+  javaImports: [
+    'foam.dao.DAO',
+    'foam.dao.ArraySink',
+    'net.nanopay.invoice.model.Invoice',
+    'foam.dao.history.HistoryRecord',
+    'static foam.mlang.MLang.*',
+    'foam.nanos.auth.AuthorizationException',
+  ],
+
+  methods: [
+    {
+      name: 'authorizeOnCreate',
+      javaCode: `
+        User user = (User) x.get("user");
+        if ( user == null ) {
+          throw new foam.nanos.auth.AuthenticationException("User not found.");
+        } else if ( ! "admin".equalsIgnoreCase(user.getGroup()) && ! "system".equalsIgnoreCase(user.getGroup())) {
+          throw new AuthorizationException();
+        }
+      `
+    },
+    {
+      name: 'authorizeOnRead',
+      javaCode: `
+        DAO invoiceDAO = (DAO) x.get("invoiceDAO");
+        User user = (User) x.get("user");
+
+        ArraySink invoiceArraySink = (ArraySink) invoiceDAO.where(
```

Comment:
Rather than doing a select where id = something, you can just use `find`.

```Java
Invoice invoice = (Invoice) invoiceDAO.find(((HistoryRecord) obj).getObjectId());
```